qcodes/instrument_drivers/Harvard/Decadac.py

```diff
@@ -115,8 +162,8 @@ def _setvoltage(self, voltage, channel):
             try:
                 self.visa_handle.read()
             except UnicodeDecodeError:
-                logging.warning(" Decadac returned nothing and did nothing. " +
-                                "Please re-run the command")
+                log.warning(" Decadac returned nothing and did nothing. " +
+                            "Please re-run the command")
                 pass
 
         if self.ramp_state:
```

Comment:
also line 173 should be self._ramp_time
also when that fails with error message: AttributeError: ("'Decadac' object and its delegates have no attribute 'ramp_time'", 'setting Decadac:ch0_voltage to 0')
it doesn't actually set the voltage to 0...